src/murmur/Messages.cpp

```diff
@@ -2104,3 +2104,32 @@ void Server::msgServerConfig(ServerUser *, MumbleProto::ServerConfig &) {
 
 void Server::msgSuggestConfig(ServerUser *, MumbleProto::SuggestConfig &) {
 }
+
+void Server::msgPluginDataTransmission(ServerUser *sender, MumbleProto::PluginDataTransmission &msg) {
+	// a client's plugin has sent us a message that we shall delegate to its receivers
+
+	if (!msg.has_data() || !msg.has_dataid()) {
+		// messages without data and/or without a data ID can't be used by the clients. Thus we don't even have to send it
+		return;
+	}
+
+	if (!msg.has_sendersession()) {
```

Comment:
Why are we allowing people to spoof their session id?

Reply:
I guess I wanted to be nice to folks that forgot to attach the sender ID to the message as I can easily check for this here.
Do you think messages without a sender ID should instead be discarded as well? :thinking:

Reply:
No, I think the if-check should be removed, the server should always set the `sendersession` to that of the true sender.
Otherwise clients cannot trust it cause everyone can use set any `sendersession` they want.

Reply:
Ah :bulb: 
That's actually a good point, yes :+1: